src/install/python/python-options/python-agent-uvicorn-docker.mdx

```diff
@@ -0,0 +1,113 @@
+
+### A. Create the directory structure [#create-directories]
+
+You'll need to insert some files in the directory structure for your application. To prepare for that, you might create a structure similar to this:
+
+```
+python-docker
+├── newrelic
+│   └── Dockerfile
+└── src
+    ├── Dockerfile
+    ├── your_app.py
+    ├── newrelic.ini
+    └── requirements.txt
+```
+
+### B. Insert the New Relic configuration file [#config-file]
+
+In the previous steps, you created a `newrelic.ini` configuration file that contains your license key and app name.
+
+Insert this configuration in the same directory as your Python app.
+
+### C. Create the requirements file [#requirements-file]
+
+Create `requirements.txt` (the list of dependencies for Python) by running this command in the directory where your application resides:
+
+```
+pip freeze > requirements.txt
+```
+
+### D. Create the base container (New Relic) [#base-container]
+
+This is the container that runs the agent.
+
+1. In the directory for the New Relic agent (for example, `newrelic`), create this Dockerfile for the base container (the file name is just `Dockerfile`):
+
+    ```
+    FROM python:3.9.14-alpine3.16
+
+    RUN pip install --no-cache-dir newrelic
+
+    ENTRYPOINT ["newrelic-admin", "run-program"]
+    ```
+
+2. Create the container with this command:
+
+    ```
+    docker build -t python_newrelic:latest .
+    ```
+
+3. Change to the directory where your app is located (for example, src)
+
+### E. Create a container for your app
+
+This is the container that will hold your application. It will work in concert with the base container. Note that it pulls in the base container in the `FROM`.
+
+1. Create the Dockerfile for your app in the same directory as your Python app.
+
+    * Change `WORKDIR` to match your directory structure.
+    * For the `CMD` line, replace the uvicorn example with the syntax from the previous step.
+
+    ```
+    FROM python_newrelic:latest
+
+    RUN apk add --no-cache bzip2-dev \
+            coreutils \
+            gcc \
+            libc-dev \
+            libffi-dev \
+            libressl-dev \
+            linux-headers
+
+        WORKDIR INSERT_THE_PATH_TO_YOUR_PYTHON_APP
+
+        COPY requirements.txt ./
+        RUN pip install --no-cache-dir -r requirements.txt
+
+        COPY . .
+
+        EXPOSE 8000
+
+        CMD ["uvicorn", "main:app", "--host", "0.0.0.0", "--port", "8000"]
+    ```
+
+2. In the directory with your Dockerfile and your app, build the container with this command:
+
+    ```
+	docker build -t my_python_api .
+    ```
+
+### F. Run the containers [#run-container]
+
+Once you have the base New Relic container and a container made for your app, you can run both with this command:
+
+```
+docker run \
+-e NEW_RELIC_LICENSE_KEY=INSERT_YOUR_LICENSE_KEY \
+-e NEW_RELIC_APP_NAME="docker-fastapi-example" \
+-p 8000:8000 -it --rm --name INSERT_NAME_TO_DISPLAY_IN_DOCKER INSERT_A_SECONDARY_NAME_FOR_YOUR_APP
```

Comment:
The 2 names here don't fit, that's not a pair for display names. One of those 2 is the image and tag, the other (after --name) is the container name to display when looked up in docker.

`docker container run [--name container_name] docker_image[:image_tag]` is the proper way to do that.